lib/process/process_test.go

```diff
@@ -324,3 +326,162 @@ type importerSuite struct {
 	backend storage.Backend
 	pack    pack.PackageService
 }
+
+func Test_reconcileLabels2(t *testing.T) {
+	type args struct {
+		currentLabels  map[string]string
+		requiredLabels map[string]string
+	}
+	tests := []struct {
+		name           string
+		args           args
+		expectedLabels map[string]string
+		needUpdate     bool
+	}{
+		{
+			name: "reconciliation mode = disabled. Different labels",
+			args: args{
+				currentLabels: map[string]string{
+					"gravitational.io/reconcile": "false",
+				},
+				requiredLabels: map[string]string{
+					"label1": "value1",
+					"label2": "value2",
+				},
+			},
+			expectedLabels: map[string]string{
+				"gravitational.io/reconcile": "false",
+			},
+			needUpdate: false,
+		},
+		{
+			name: "reconciliation mode = enabled. Different labels",
+			args: args{
+				currentLabels: map[string]string{
+					"gravitational.io/reconcile": "true",
+					"label1":                     "1",
+					"label2":                     "1",
+				},
+				requiredLabels: map[string]string{
+					"label1": "value1",
+					"label2": "value2",
+				},
+			},
+			expectedLabels: map[string]string{
+				"gravitational.io/reconcile": "true",
+				"label1":                     "value1",
+				"label2":                     "value2",
+			},
+			needUpdate: true,
+		},
+		{
+			name: "reconciliation mode = enabled. Same labels",
+			args: args{
+				currentLabels: map[string]string{
+					"gravitational.io/reconcile": "true",
+					"label1":                     "value1",
+					"label2":                     "value2",
+				},
+				requiredLabels: map[string]string{
+					"label1": "value1",
+					"label2": "value2",
+				},
+			},
+			expectedLabels: map[string]string{
+				"gravitational.io/reconcile": "true",
+				"label1":                     "value1",
+				"label2":                     "value2",
+			},
+			needUpdate: false,
+		},
+		{
+			name: "reconciliation mode = EnsureExists. Different labels",
+			args: args{
+				currentLabels: map[string]string{
+					"gravitational.io/reconcile": "EnsureExists",
+				},
+				requiredLabels: map[string]string{
+					"label1": "value1",
+					"label2": "value2",
+				},
+			},
+			expectedLabels: map[string]string{
+				"gravitational.io/reconcile": "EnsureExists",
+				"label1":                     "value1",
+				"label2":                     "value2",
+			},
+			needUpdate: true,
+		},
+		{
+			name: "reconciliation mode = EnsureExists. Different value of labels",
```

Comment:
Maybe add a hint at the intent (to emphasize the difference between `Enabled` and `EnsureExists`):
```
name: reconciliation mode = EnsureExists. Different value of labels and no change is expected.
```

Reply:
Sure